n2kMapper.js

```diff
@@ -15,33 +15,56 @@ function N2kMapper (options) {
   this.state = {}
 
   this.on('n2kRequestMetadata', (src) => {
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126996
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 126998
-    })
-    this.emit('n2kOut', {
-      "pgn": 59904,
-      "dst": src,
-      "PGN": 60928
-    })
+    if ( src === 255 ) {
+      return
+    }
+    this.requestMetaData(src, 126996)
+    this.requestMetaData(src, 126998)
+    this.requestMetaData(src, 60928)
   })
 }
 
+N2kMapper.prototype.requestMetaData = function(src, pgn) {
+  const reqPgn = {
+    "pgn": 59904,
+    "dst": src,
+    "PGN": pgn
+  }
+  let retries = 5
+  debug(`requesting pgn ${pgn} from src ${src}`)
+  this.emit('n2kOut', reqPgn)
+  let interval = setInterval(() => {
+    if ( retries-- === 0 ) {
+      debug(`did not get meta pgn ${pgn} for src ${src}`)
+      clearInterval(interval)
+      this.emit('n2kSourceMetadataTimeout', pgn, src)
+    } else if ( this.state[src] && this.state[src].metaPGNsReceived && this.state[src].metaPGNsReceived[pgn] ) {
```

Comment:
Retry works only for the first time, afterwards we already have metadata and there is no retry on subsequent requests.

Reply:
I didn't want to retry anymore, I figure if it didn't respond to the retries it never will. Probably means the device does not support this.

Reply:
I still think that emitting `requestMetaData` multiple times should work consistently so that subsequent events also cause retries, no matter who is emitting it (in #185 we clear the meta ourselves before requesting metadata, but what if it is somebody else?).

What if we kept a timestamp of when we got meta and the retry mechanism relied on that, not the existence of meta?

Reply:
Not sure I understand...

Reply:
See #188 - would that work? So that the interval checks that we have received metadata after this metadata request was sent, not just that we have received it sometime in the past.